microservices-orders/docs/index.rst

```diff
@@ -0,0 +1,578 @@
+The result of each validation is pushed through a separate topic, Order Validations, so that we retain the `single writer’ status of the Orders Service —> Orders Topic (there are several options for managing consistency in event collaboration, discussed in Ben Stopford's `book <https://www.confluent.io/designing-event-driven-systems>`__).
+The results of the various validation checks are aggregated back in the Order Service (Validation Aggregator) which then moves the order to a Validated or Failed state, based on the combined result.
+
+To allow users to GET any order, the Orders Service creates a queryable materialized view (embedded inside the Orders Service), using a state store in each instance of the service, so any Order can be requested historically. Note also that the Orders Service can be scaled out over a number of nodes, so GET requests must be routed to the correct node to get a certain key. This is handled automatically using the Interactive Queries functionality in Kafka Streams.
+
+The Orders Service also includes a blocking HTTP GET so that clients can read their own writes. In this way we bridge the synchronous, blocking paradigm of a Restful interface with the asynchronous, non-blocking processing performed server-side.
+
+Finally there is a very simple email service.
+
+.. figure:: images/microservices-exercises-combined.jpg
+    :alt: image
+
+All the services are client applications written in Java, and they use the Kafka Streams API.
+The java source code for these microservices are in the `kafka-streams-examples repo <https://github.com/confluentinc/kafka-streams-examples/tree/5.0.1-post/src/main/java/io/confluent/examples/streams/microservices>`__.
+
+Summary of services and the topics they consume from and produce to:
+
++-------------------------------------+-----------------------------------+-----------------------+
+| Service                             | Consumes From                     | Produces To           |
++=====================================+===================================+=======================+
+| InventoryService                    | `orders`, `warehouse-inventory`   | `order-validations`   |
++-------------------------------------+-----------------------------------+-----------------------+
+| FraudService                        | `orders`                          | `order-validations`   |
++-------------------------------------+-----------------------------------+-----------------------+
+| OrderDetailsService                 | `orders`                          | `order-validations`   |
++-------------------------------------+-----------------------------------+-----------------------+
+| ValidationsAggregatorService        | `order-validations`, `orders`     | `orders`              |
```

Comment:
A couple of the services i.e. `ValidationsAggregatorService` and `PostOrdersAndPayments` and `AddInventory` are not explicitly explained in the above text. Better add a few words each.

Reply:
Good catch